p4rt_app/p4runtime/p4runtime_impl.cc

```diff
@@ -13,52 +13,60 @@
 // limitations under the License.
 #include "p4rt_app/p4runtime/p4runtime_impl.h"
 
-#include <endian.h>
-
-#include <iomanip>
-#include <iostream>
-#include <sstream>
+#include <memory>
 #include <stdexcept>
 #include <string>
-#include <type_traits>
+#include <vector>
 
 #include "absl/memory/memory.h"
 #include "absl/status/status.h"
 #include "absl/status/statusor.h"
 #include "absl/strings/str_cat.h"
-#include "absl/strings/substitute.h"
+#include "absl/strings/str_format.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
+#include "boost/bimap.hpp"
 #include "glog/logging.h"
 #include "google/protobuf/descriptor.h"
 #include "google/protobuf/util/message_differencer.h"
 #include "google/rpc/code.pb.h"
 #include "grpcpp/impl/codegen/status.h"
-#include "gutil/collections.h"
+#include "grpcpp/support/status.h"
+#include "gutil/proto.h"
 #include "gutil/status.h"
 #include "p4/config/v1/p4info.pb.h"
+#include "p4/v1/p4runtime.pb.h"
 #include "p4_constraints/backend/constraint_info.h"
 #include "p4_constraints/backend/interpreter.h"
 #include "p4_pdpi/ir.h"
 #include "p4_pdpi/ir.pb.h"
-#include "p4_pdpi/utils/annotation_parser.h"
-#include "p4_pdpi/utils/ir.h"
 #include "p4rt_app/p4runtime/ir_translation.h"
 #include "p4rt_app/p4runtime/p4info_verification.h"
-#include "p4rt_app/sonic/adapters/consumer_notifier_adapter.h"
-#include "p4rt_app/sonic/adapters/db_connector_adapter.h"
-#include "p4rt_app/sonic/adapters/producer_state_table_adapter.h"
+#include "p4rt_app/p4runtime/packetio_helpers.h"
 #include "p4rt_app/sonic/app_db_acl_def_table_manager.h"
 #include "p4rt_app/sonic/app_db_manager.h"
-#include "p4rt_app/sonic/packetio_port.h"
+#include "p4rt_app/sonic/packetio_interface.h"
+#include "p4rt_app/sonic/redis_connections.h"
 #include "p4rt_app/sonic/response_handler.h"
+#include "p4rt_app/sonic/state_verification.h"
 #include "p4rt_app/utils/status_utility.h"
 #include "p4rt_app/utils/table_utility.h"
-#include "sai_p4/fixed/ids.h"
-#include "sai_p4/fixed/roles.h"
 
 namespace p4rt_app {
 namespace {
 
-using ::google::protobuf::util::MessageDifferencer;
+grpc::Status EnterCriticalState(const std::string& message) {
+  // TODO: report critical state somewhere an don't crash the process.
```

Comment:
When is critical state infra to be upstreamed ? 

For the time being, can we make it a WARNING and use a global var to know if in critical state or not ?

Reply:
I'm not sure what the timelines are for critical state. We could do a global variable, but the further away we take it the harder merges and up-streaming becomes.